src/utils/api.js

```diff
@@ -128,16 +125,30 @@ function createAPIError(response) {
   return error;
 }
 
-function attachJSON(response) {
-  return response.json()
-  .then(data => {
-    response.data = data;
-    return response;
-  })
-  .catch(() => {
-    response.data = {};
-    return response;
-  })
+function attachData(response) {
+  let contentType = response.headers.get('Content-Type') || ''
+  if (contentType.includes('text/csv')){
+    return response.text()
+      .then(text => {
+        return text;
+      })
+  } else{
+    return response.json()
+      .then(data => {
+        response.data = data;
+        return response;
+      })
+      .catch(() => {
+        response.data = {};
+        return response;
+      })
+      .then(response => {
+        if(response.ok) {
+          return response;
+        }
+        return Promise.reject(createAPIError(response));
+      })
+  }
```

Comment:
Why is the `response.ok` check in `attachData` now? And why is it only done for JSON?